frontend/src/lib/lemon-ui/LemonSelectMultiple/LemonSelectMultiple.tsx

```diff
@@ -1,121 +1,264 @@
-    labelInValue?: boolean
-    onSearch?: (value: string) => void
-    onFocus?: () => void
-    onBlur?: () => void
-    filterOption?: boolean
-    mode?: 'single' | 'multiple' | 'multiple-custom'
-    onChange?: ((newValue: string) => void) | ((newValue: string[]) => void)
+    disableFiltering?: boolean
+    mode?: 'single' | 'multiple'
+    allowCustomValues?: boolean
+    onChange?: (newValue: string[]) => void
+    onInputChange?: (newValue: string) => void
     'data-attr'?: string
 }
 
-export type LabelInValue = { value: string; label: ReactNode }
-
 export function LemonSelectMultiple({
+    placeholder,
+    options = [],
     value,
-    options,
-    disabled,
     loading,
-    placeholder,
-    labelInValue,
     onChange,
-    onSearch,
-    onFocus,
-    onBlur,
-    filterOption = true,
-    mode = 'single',
-    selectClassName,
+    onInputChange,
+    mode,
+    disabled,
+    disableFiltering = false,
+    allowCustomValues = false,
     ...props
 }: LemonSelectMultipleProps): JSX.Element {
-    const optionsAsList: LemonSelectMultipleOptionItem[] = Array.isArray(options)
-        ? options
-        : Object.entries(options || {}).map(([key, option]) => ({
-              key: key,
-              ...option,
-          }))
-
-    const antOptions = optionsAsList.map((option) => ({
-        key: option.key,
-        value: option.key,
-        label: option.labelComponent || option.label,
-        labelString: option.label,
-    }))
+    const [showPopover, setShowPopover] = useState(false)
+    const [inputValue, setInputValue] = useState('')
+    const popoverFocusRef = useRef<boolean>(false)
+    const inputRef = useRef<HTMLInputElement>(null)
+    const [selectedIndex, setSelectedIndex] = useState(0)
+    const values = value ?? []
+
+    const visibleOptions = useMemo(() => {
+        const res: LemonSelectMultipleOption[] = []
+        const customValues = [...values]
+
+        options.forEach((option) => {
+            // Remove from the custom values list if it's in the options
+
+            if (customValues.includes(option.key)) {
+                customValues.splice(customValues.indexOf(option.key), 1)
+            }
+
+            // Check for filtering
+            if (inputValue && !disableFiltering && !option.label.toLowerCase().includes(inputValue.toLowerCase())) {
+                return
+            }
+
+            res.push(option)
+        })
+
+        // Custom values are always shown before the list
+        if (customValues.length) {
+            customValues.forEach((value) => {
+                res.unshift({ key: value, label: value })
+            })
+        }
+
+        // Finally we show the input value if custom values are allowed and it's not in the list
+        if (allowCustomValues && inputValue && !values.includes(inputValue)) {
+            res.unshift({ key: inputValue, label: inputValue })
+        }
+
+        return res
+    }, [options, inputValue, ...values])
```

Comment:
Looks like this might cause some console errors
<img width="1234" alt="Screenshot 2024-03-15 at 15 11 46" src="https://github.com/PostHog/posthog/assets/6685876/08abd7ce-5700-468f-a835-2d7f98750244">